### PR TITLE
feat(dashboards): Widget Viewer total result count in footer

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Adds a total result count for widgets in the widget viewer footer.
For Discover widgets, this is achieved by calling `events-meta` on mount to get the total count.
For Issue widgets, this value is already returned in the headers of the issues api request.
image

### CODE REVIEW

File: static/app/components/modals/widgetViewerModal.tsx [hunk 1]
```diff
@@ -83,6 +88,29 @@ const MemoizedWidgetCardChartContainer = React.memo(
...
+    const total = await fetchTotalCount(
+      api,
+      organization.slug,
+      eventView.getEventsAPIPayload(location)
+    );
+    return total.toLocaleString();
+  } catch (err) {
+    Sentry.captureException(err);
+  }
+  return undefined;
```
Comment (priscilawebdev) [inline static/app/components/modals/widgetViewerModal.tsx:111]: should we place this return inside the catch?

File: static/app/components/modals/widgetViewerModal.tsx [hunk 2]
```diff
@@ -491,6 +533,20 @@ function WidgetViewerModal(props: Props) {
       </StyledHeader>
       <Body>{renderWidgetViewer()}</Body>
       <StyledFooter>
+        <TotalResultsContainer>
+          {totalResults &&
+            (widget.widgetType === WidgetType.ISSUE ? (
+              <span>
+                <b>{t('Total Issues: ')}</b>
+                {totalResults === '1000' ? '1000+' : totalResults}
```
Comment (priscilawebdev) [inline static/app/components/modals/widgetViewerModal.tsx:541]: you could also have written as following

```suggestion
               {tct('[description:Total Issues:] [total].', {
                  description: <strong />,
                  total: totalResults === '1000' ? '1000+' : totalResults,
                })}
```

but it looks good to me

File: tests/js/spec/components/modals/widgetViewerModal.spec.tsx [hunk 3]
```diff
@@ -145,14 +161,17 @@ describe('Modals -> WidgetViewerModal', function () {
     });
 
     it('renders area chart', async function () {
+      await renderModal({initialData, widget: mockWidget});
       expect(await screen.findByText('echarts mock')).toBeInTheDocument();
```
Comment (priscilawebdev) [inline tests/js/spec/components/modals/widgetViewerModal.spec.tsx:165]: is the second `await` needed?

**Comment by priscilawebdev** [comment]
image
 should we check this issue on smaller devices?

**Comment by edwardgou-sentry** [comment]
> should we break the lines here on smaller devices?

I updated the styling a bit to better fit smaller devices:
image

**Review by priscilawebdev — APPROVED** [review]
lgtm

**Pull request merged**